### ISSUE TITLE
[Feature Request]: Overlay Images within Krypton Toolkit controls (Not Taskbar)

### ISSUE THREAD

**Issue by PWagner1** [Krypton-Suite/Standard-Toolkit#1205]
Is it possible to support 'overlay images' On Krypton Control Images?
![image](https://github.com/Krypton-Suite/Standard-Toolkit/assets/949607/b2d24e89-9a8f-4564-9f54-6c95bb1a8700)

The position of the indicator has to be programmable (designer) as image above and below show (all 4 corners maybe?)
![image](https://github.com/Krypton-Suite/Standard-Toolkit/assets/2418812/946c83b9-3878-45af-b796-801a586d0c5f)

- [ ] Add K Control
- [ ] Add / extend a demo app

**Comment by Smurf-IV** [comment]
I thought that was a Windows Task bar thing ?
But I'm assuming you would want this to be added to "Normal Buttons" ??
And then also to any thing else (Like Error Indicators ?)

**Comment by Wagnerp** [comment]
> I thought that was a Windows Task bar thing ? But I'm assuming you would want this to be added to "Normal Buttons" ?? And then also to any thing else (Like Error Indicators ?)

Yes, I was thinking about buttons specifically when doing the UAC shield stuff, rather than losing the original image entirely. Also using is as a indicator would also be good.

**Comment by Smurf-IV** [comment]
https://learn.microsoft.com/en-us/windows/win32/api/shobjidl_core/nf-shobjidl_core-itaskbarlist3-setoverlayicon
https://docwiki.embarcadero.com/Libraries/Sydney/en/Vcl.Taskbar.TCustomTaskbar.OverlayIcon

Should be as easy as using it in html:
https://www.w3schools.com/howto/howto_css_image_overlay_icon.asp

**Comment by Smurf-IV** [comment]
Trying to do the Overlay via "Windows Shell API's" is overkill:
I think it could be as simple as replacing the icon with another that has the Overlay "Added / blended"; then lets windows do it's thing normally ;-)

**Comment by Wagnerp** [comment]
> Trying to do the Overlay via "Windows Shell API's" is overkill: I think it could be as simple as replacing the icon with another that has the Overlay "Added / blended"; then lets windows do it's thing normally ;-)

I think that's better, also for the messagebox, if the user sets a custom icon, place a second picturebox over it with the overlay icon?

**Comment by Smurf-IV** [comment]
> > Trying to do the Overlay via "Windows Shell API's" is overkill: I think it could be as simple as replacing the icon with another that has the Overlay "Added / blended"; then lets windows do it's thing normally ;-)
> 
> I think that's better, also for the messagebox, if the user sets a custom icon, place a second picturebox over it with the overlay icon?

No.. This will lead to all sorts of complex mouse over issues.
If the button has an icon, AND the Dev has stated to "Switch on Overlay", then perform icon combo and and set button image to combo. When Dev Turns off the icon overlay, then restore the the original icon.

The above solves the Icons / images in existing KToolkit items.
But there is also the issue of the "Error Icon overlay" that can be anywhere within the form that owns the parent control - See coming related Feature request...

**Comment by Wagnerp** [comment]
@Smurf-IV I can possibly 'raid' both Windows & VS for some default overlays, i.e. warning, error etc. Though these are likely to be docked to the bottom right.

**Comment by Wagnerp** [comment]
> The Overlay Icon (Currently) has to be 16x16 pixels, so it should "Fit" into any 32x32 that is displayed on an image icon.

You can resize them to 32 x 32 with no problem, but a scaler would be useful for custom icons

![OverlayOK](https://github.com/Krypton-Suite/Standard-Toolkit/assets/949607/2a7e3274-0008-41c2-b31a-2f002f8d8793)

**Comment by PWagner1** [comment]
Moving to V100
